File: src/test/java/com/polygloat/AbstractTransactionalTest.java
package com.polygloat;

import com.polygloat.repository.LanguageRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.testng.AbstractTransactionalTestNGSpringContextTests;
import org.springframework.test.context.transaction.TestTransaction;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;

@Transactional
public abstract class AbstractTransactionalTest extends AbstractTransactionalTestNGSpringContextTests {
    @Autowired
    protected EntityManager entityManager;

    @Autowired
    protected LanguageRepository languageRepository;

    protected void commitTransaction() {
        TestTransaction.flagForCommit();
        entityManager.flush();
        TestTransaction.end();
        TestTransaction.start();
        entityManager.clear();
    }

}

File: backend/data/src/main/kotlin/io/tolgee/testing/AbstractTransactionalTest.kt
package io.tolgee.testing

import io.tolgee.repository.LanguageRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.testng.AbstractTransactionalTestNGSpringContextTests
import org.springframework.test.context.transaction.TestTransaction
import org.springframework.transaction.annotation.Transactional
import javax.persistence.EntityManager

@Transactional
abstract class AbstractTransactionalTest : AbstractTransactionalTestNGSpringContextTests() {
    @Autowired
    lateinit protected var entityManager: EntityManager

    @Autowired
    lateinit protected var languageRepository: LanguageRepository

    protected fun commitTransaction() {
        TestTransaction.flagForCommit()
        entityManager.flush()
        TestTransaction.end()
        TestTransaction.start()
        entityManager.clear()
    }
}
